### PR TITLE
Ucsc update

### PR DESCRIPTION
**Pull request by DLBPointon**
Updating UCSC modules
3 needed updating from 355 to 447 and relevant md5sums changed.
Removed unnecessary stub test from yml that wasn't being used.
Added stub to 1 module.

## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] Emit the `versions.yml` file.
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/ucsc/bedgraphtobigwig/main.nf [hunk 1]
```diff
@@ -3,10 +3,10 @@ process UCSC_BEDGRAPHTOBIGWIG {
     label 'process_single'
 
     // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
-    conda "bioconda::ucsc-bedgraphtobigwig=377"
+    conda "bioconda::ucsc-bedgraphtobigwig=445=h954228d_0"
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/bedgraphtobigwig/main.nf:6]:
```suggestion
    conda "bioconda::ucsc-bedgraphtobigwig=445"
```

File: modules/nf-core/ucsc/bedgraphtobigwig/main.nf [hunk 2]
```diff
@@ -22,7 +22,7 @@ process UCSC_BEDGRAPHTOBIGWIG {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '445=h954228d_0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/bedgraphtobigwig/main.nf:25]:
```suggestion
    def VERSION = '445' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

File: modules/nf-core/ucsc/bedgraphtobigwig/main.nf [hunk 3]
```diff
@@ -37,7 +37,7 @@ process UCSC_BEDGRAPHTOBIGWIG {
 
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '445=h954228d_0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/bedgraphtobigwig/main.nf:40]:
```suggestion
    def VERSION = '445' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

File: modules/nf-core/ucsc/bedtobigbed/main.nf [hunk 4]
```diff
@@ -3,10 +3,10 @@ process UCSC_BEDTOBIGBED {
     label 'process_single'
 
     // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
-    conda "bioconda::ucsc-bedtobigbed=377"
+    conda "bioconda::ucsc-bedtobigbed=447=h954228d_0"
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/bedtobigbed/main.nf:6]:
```suggestion
    conda "bioconda::ucsc-bedtobigbed=447"
```

File: modules/nf-core/ucsc/bedtobigbed/main.nf [hunk 5]
```diff
@@ -24,7 +24,7 @@ process UCSC_BEDTOBIGBED {
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     def as_option = autosql ? "-as=${autosql}" : ""
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '447=h954228d_0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/bedtobigbed/main.nf:27]:
```suggestion
    def VERSION = '447' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

File: modules/nf-core/ucsc/bedtobigbed/main.nf [hunk 6]
```diff
@@ -41,7 +41,7 @@ process UCSC_BEDTOBIGBED {
 
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '447=h954228d_0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/bedtobigbed/main.nf:44]:
```suggestion
    def VERSION = '447' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

File: modules/nf-core/ucsc/gtftogenepred/main.nf [hunk 7]
```diff
@@ -2,10 +2,10 @@ process UCSC_GTFTOGENEPRED {
     tag '${meta.id}'
     label 'process_low'
 
-    conda "bioconda::ucsc-gtftogenepred=377"
+    conda "bioconda::ucsc-gtftogenepred=447=h954228d_0"
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/gtftogenepred/main.nf:5]:
```suggestion
    conda "bioconda::ucsc-gtftogenepred=447"
```

File: modules/nf-core/ucsc/gtftogenepred/main.nf [hunk 8]
```diff
@@ -22,7 +22,7 @@ process UCSC_GTFTOGENEPRED {
     def args = task.ext.args ?: ''
     def gen_refflat = args.contains("-genePredExt") && args.contains("-geneNameAsName2") ? "true" : "false"
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '447=h954228d_0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/gtftogenepred/main.nf:25]:
```suggestion
    def VERSION = '447' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

File: modules/nf-core/ucsc/gtftogenepred/main.nf [hunk 9]
```diff
@@ -41,7 +41,7 @@ process UCSC_GTFTOGENEPRED {
 
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377'
+    def VERSION = '447=h954228d_0'
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/gtftogenepred/main.nf:44]:
```suggestion
    def VERSION = '447'
```

File: modules/nf-core/ucsc/wigtobigwig/main.nf [hunk 10]
```diff
@@ -3,10 +3,10 @@ process UCSC_WIGTOBIGWIG {
     label 'process_single'
 
     // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
-    conda "bioconda::ucsc-wigtobigwig=377"
+    conda "bioconda::ucsc-wigtobigwig=447=h2a80c09_1"
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/wigtobigwig/main.nf:6]:
```suggestion
    conda "bioconda::ucsc-wigtobigwig=447"
```

File: modules/nf-core/ucsc/wigtobigwig/main.nf [hunk 11]
```diff
@@ -22,7 +22,7 @@ process UCSC_WIGTOBIGWIG {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '447=h2a80c09_1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/wigtobigwig/main.nf:25]:
```suggestion
    def VERSION = '447' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

File: modules/nf-core/ucsc/wigtobigwig/main.nf [hunk 12]
```diff
@@ -38,7 +38,7 @@ process UCSC_WIGTOBIGWIG {
 
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def VERSION = '447=h2a80c09_1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (nvnieuwk) [inline modules/nf-core/ucsc/wigtobigwig/main.nf:41]:
```suggestion
    def VERSION = '447' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```

**Review by nvnieuwk — COMMENTED** [review]
Only containers need to have the tag in their versioning :)

**Review by muffato — COMMENTED** [review]
Hi. I couldn't find it in the [official documentation about modules](https://nf-co.re/docs/contributing/modules), but my understanding (based on the [behaviour of the `nf-core` command](https://github.com/nf-core/tools/blob/master/nf_core/modules/test_yml_builder.py#L245-L248)) is that `versions.yml` should be listed _without_ a MD5 checksum.

**Comment by nvnieuwk** [comment]
yes @muffato is right! It shouldn't be there if you use the latest versions of the `nf-core` tooling to generate this

**Comment by DLBPointon** [comment]
Ok, thanks both @muffato @nvnieuwk 
I've made the changes you've suggested. I also tried to be helpful and add the path for versions into liftover. However as you can see it is now throwing errors due to an output file being empty. 
I haven't changed anything in the file so it works as it did before i stuck my nose in.

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Pull request merged**